From a65613bf0a507a226c172bb6f503e27532efb0fa Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Thu, 08 Sep 2011 17:01:35 +0000
Subject: GnomeWallClock: Remove the date/time source on dispose

We don't do it in finalize since the dispose handler needs the objects
that are being disposed.

https://bugzilla.gnome.org/show_bug.cgi?id=658583
---
diff --git a/libgnome-desktop/gnome-wall-clock.c b/libgnome-desktop/gnome-wall-clock.c
index 49437d8..167619d 100644
--- a/libgnome-desktop/gnome-wall-clock.c
+++ b/libgnome-desktop/gnome-wall-clock.c
@@ -84,6 +84,11 @@ gnome_wall_clock_dispose (GObject *object)
 {
 	GnomeWallClock *self = GNOME_WALL_CLOCK (object);
 
+	if (self->priv->clock_update_id) {
+		g_source_remove (self->priv->clock_update_id);
+		self->priv->clock_update_id = 0;
+	}
+
 	if (self->priv->tz_monitor != NULL) {
 		g_object_unref (self->priv->tz_monitor);
 		self->priv->tz_monitor = NULL;
--
cgit v0.9.0.2
